python/python/glide/async_commands/standalone_commands.py

```diff
@@ -262,6 +262,59 @@ async def function_load(self, library_code: str, replace: bool = False) -> str:
             ),
         )
 
+    async def function_list(
+        self, library_name_pattern: Optional[str] = None, with_code: bool = False
+    ) -> List[
+        Mapping[
+            bytes,
+            List[
+                Mapping[
+                    bytes, Union[bytes, List[Mapping[bytes, Union[bytes, Set[bytes]]]]]
+                ]
+            ],
+        ]
+    ]:
+        """
+        Returns information about the functions and libraries.
+
+        See https://valkey.io/commands/function-list/ for more details.
+
+        Args:
+            library_name_pattern (Optional[str]):  A wildcard pattern for matching library names.
+            with_code (bool): Specifies whether to request the library code from the server or not.
+
+        Returns:
+            List[Mapping[bytes, Union[bytes, List[Mapping[bytes, Union[bytes, Set[bytes]]]]]]]: Info about all
+                libraries and their functions.
+
+        Examples:
+            >>> response = await client.function_list("myLib?_backup", True)
+                [{
+                    b"library_name": b"myLib5_backup",
+                    b"engine": b"LUA",
+                    b"functions": [{
+                        b"name": func_name.encode(),
+                        b"description": None,
+                        b"flags": {b"no-writes"},
+                    }],
+                    b"library_code": b"#!lua name=mylib \n redis.register_function('myfunc', function(keys, args) return args[1] end)"
+                }]
+
+        Since: Redis 7.0.0.
+        """
+        args = []
+        if library_name_pattern is not None:
+            args.extend(["LIBRARYNAME", library_name_pattern])
+        if with_code:
+            args.append("WITHCODE")
+        return cast(
+            List[Mapping[bytes, Any]],
```

Comment:
If we decide to keep the long return type above, update here:
```suggestion
            List[Mapping[bytes, Union[bytes, List[Mapping[bytes, Union[bytes, Set[bytes]]]]]]]
```